src/test/java/org/codehaus/mojo/versions/DependencyUpdatesReportTest.java

```diff
@@ -0,0 +1,234 @@
+                                            List<Dependency> dependencyManagement )
+        {
+            mockPlexusComponents();
+
+            project = new MavenProject();
+            project.setDependencies( dependencies );
+
+            project.setOriginalModel( new Model() );
+            project.getOriginalModel().setDependencyManagement( new DependencyManagement() );
+            project.getOriginalModel().getDependencyManagement().setDependencies( originalDependencyManagement );
+
+            project.getModel().setDependencyManagement( new DependencyManagement() );
+            project.getModel().getDependencyManagement().setDependencies( dependencyManagement );
+
+            artifactMetadataSource = mock( ArtifactMetadataSource.class );
+            try
+            {
+                when( artifactMetadataSource.retrieveAvailableVersions( any( Artifact.class ), any(), any() ) ).then(
+                        invocation -> {
+                            Artifact artifact = invocation.getArgument( 0 );
+                            if ( "artifactA".equals( artifact.getArtifactId() ) && "1.0.0".equals(
+                                    artifact.getVersion() ) )
+                            {
+                                return Arrays.asList( new DefaultArtifactVersion( artifact.getVersion() ),
+                                        new DefaultArtifactVersion( "2.0.0" ) );
+                            }
+                            if ( "artifactB".equals( artifact.getArtifactId() ) && "1.0.0".equals(
+                                    artifact.getVersion() ) )
+                            {
+                                return Arrays.asList( new DefaultArtifactVersion( artifact.getVersion() ),
+                                        new DefaultArtifactVersion( "1.1.0" ) );
+                            }
+                            return Collections.singletonList( new DefaultArtifactVersion( artifact.getVersion() ) );
+                        } );
+            }
+            catch ( ArtifactMetadataRetrievalException e )
+            {
+                throw new RuntimeException( e );
+            }
+        }
+    }
+
+    private Dependency dependencyOf( String artifactId )
+    {
+        return DependencyBuilder.dependencyWith( "groupA", artifactId, "1.0.0", "default", "pom", SCOPE_COMPILE );
+    }
+
+    @Test
+    public void testOnlyUpgradableDependencies() throws IOException, MavenReportException
+    {
+        OutputStream os = new ByteArrayOutputStream();
+        SinkFactory sinkFactory = new Xhtml5SinkFactory();
+        new TestDependencyUpdatesReport(
+                Arrays.asList( dependencyOf( "artifactA" ), dependencyOf( "artifactB" ),
+                        dependencyOf( "artifactC" ) ), EMPTY_LIST, EMPTY_LIST )
+        {
+            {
+                onlyUpgradable = true;
+            }
+        }.generate( sinkFactory.createSink( os ), sinkFactory, Locale.getDefault() );
+
```

Comment:
For me will be more readable something like:

```java
new TestDependencyUpdatesReport()
   .withDependencies( asList( ... ) ) 
   .withDependenciesManagment( asList( ... ) )
...
  .withOnlyUpgradable( true )
 .generate ( ... );
  ```

`withXX` methods in TestDependencyUpdatesReport